File: fix_fetch.py
import os
import re

filepath = 'app/admin/page.tsx'
with open(filepath, 'r') as f:
    content = f.read()

lines = content.split('\n')
new_lines = []
for line in lines:
    if 'fetch(`https://translate.googleapis.com' in line:
        new_lines.append(line)
        continue
    if 'await fetch(' in line:
        line = line.replace('await fetch(', 'await fetchWithAuth(')
    elif 'fetch(' in line and 'io(' not in line and 'fetchWithAuth' not in line and 'fetchData' not in line:
        if 'fetch(`${apiUrl}' in line:
            line = line.replace('fetch(', 'fetchWithAuth(')
    new_lines.append(line)

with open(filepath, 'w') as f:
    f.write('\n'.join(new_lines))

File: fix_fetch.js
const fs = require('fs');
let content = fs.readFileSync('app/admin/page.tsx', 'utf8');

// Remove the previously added window.fetch interceptor safely
content = content.replace(/if \(!\(window as any\)\.__fetchIntercepted\) \{[\s\S]*?\(window as any\)\.__fetchIntercepted = true;\s*\}/g, "");

content = content.replace(/\bfetch\(/g, 'fetchWithAuth(');

const fetchWithAuthDef = `
  const fetchWithAuth = async (input: RequestInfo | URL, init?: RequestInit) => {
    let urlStr = "";
    if (typeof input === "string") urlStr = input;
    else if (input instanceof URL) urlStr = input.toString();
    else if (input instanceof Request) urlStr = input.url;

    const isApiUrl = urlStr.includes(apiUrl);
    const isLogin = urlStr.includes("/api/admin/login");

    if (isApiUrl && !isLogin) {
      const token = localStorage.getItem("admin_token");
      init = init || {};
      init.headers = {
        ...init.headers,
        "Authorization": token ? \`Bearer \${token}\` : ""
      };
    }
    return window.fetch(input, init);
  };
`;

content = content.replace(
  /const apiUrl = process\.env\.NEXT_PUBLIC_API_URL \|\| "http:\/\/localhost:5001";/g,
  `const apiUrl = process.env.NEXT_PUBLIC_API_URL || "http://localhost:5001";\n${fetchWithAuthDef}`
);

fs.writeFileSync('app/admin/page.tsx', content, 'utf8');
console.log('Fixed fetches');
